app/src/main/java/com/nilhcem/blenamebadge/ui/fragments/MainTextDrawableFragment.kt

```diff
@@ -314,35 +314,41 @@ class MainTextDrawableFragment : BaseFragment() {
         return inflater.inflate(R.layout.fragment_main_text, container, false)
     }
 
-    class StoreAsync(private val filename: String, private val json: String, private val context: WeakReference<Context>, private val listener: PreviewChangeListener) : AsyncTask<Void, Void, Boolean>() {
+    inner class StoreAsync(private val filename: String, private val json: String, private val context: WeakReference<Context>, private val listener: PreviewChangeListener) : AsyncTask<Void, Void, Boolean>() {
         override fun doInBackground(vararg params: Void?): Boolean? {
-            return (!StorageUtils.checkIfFilePresent(filename))
+            return when (StorageUtils.checkIfFilePresent(filename)) {
+                true -> {
+                    saveFile()
+                    true
+                }
+                false -> showFileOverrideDialog()
```

Comment:
very vague, wont work.

Reply:
From what I understand, you made the StoreAsync class to make the action of saving a new configuration in a different thread. Is it correct?

Reply:
yep correct

Reply:
well then currently, the function saveFile() is called in onPostExecute(), which is in the main thread. What I did was to move in into the different thread and then check if action saving is successful then   call listener.updateSavedList()

Reply:
yeah you're completely right on that. But you're calling `showFileOverrideDialog()` in doInBackground. It just wont work, it is not in the main thread and hence will not have the application context with it

Reply:
and also, the logic is the opposite, you should save the file if it did not exist. It is the opposite right now. Have a look there

Reply:
oh right that's true, I'll send update later